app/models/file_set.rb

```diff
@@ -32,4 +32,10 @@ def build_structure
   def files_metadata
     files.collect { |f| { id: f.id, label: f.label.first, external_file_uri: f.external_file_uri.first } }
   end
+
+  def to_solr(solr_doc = {})
```

Comment:
Did you give up on trying to put this in the indexer?

Reply:
I tested with that part removed and it worked fine, so I'm guessing it's not needed. It seems to work as well if I put it there too. Do you think if anything might be missed with that removed from the indexer?

Reply:
I think if you put it in the indexer, then you don't need it here. I think it would be better to keep all the indexing code in the indexer.